frontend/lib/tests/signup-intent.test.tsx

```diff
@@ -0,0 +1,33 @@
+import { validateDjangoChoices } from "../common-data";
+import { OnboardingInfoSignupIntent } from "../queries/globalTypes";
+import { signupIntentFromOnboardingInfo, DEFAULT_SIGNUP_INTENT_CHOICE, getOnboardingRouteForIntent } from "../signup-intent";
+import { AppTesterPal } from "./app-tester-pal";
+
+type SignupIntentDjangoChoice = [OnboardingInfoSignupIntent, string];
+
+const SIGNUP_INTENT_CHOICES = require('../../../common-data/signup-intent-choices.json') as SignupIntentDjangoChoice[];
+
+test('OnboardingInfoSignupIntent has valid choices', () => {
+  for (let choice in OnboardingInfoSignupIntent) {
+    validateDjangoChoices(SIGNUP_INTENT_CHOICES, [choice, OnboardingInfoSignupIntent[choice]]);
+  }
+});
+
+test('signupIntentFromOnboardingInfo() works', () => {
+  expect(signupIntentFromOnboardingInfo(null)).toStrictEqual(DEFAULT_SIGNUP_INTENT_CHOICE);
+  expect(signupIntentFromOnboardingInfo({ signupIntent: 'boop' } as any)).toBe('boop');
+});
+
+describe('getOnboardingRouteForIntent()', () => {
+  afterEach(AppTesterPal.cleanup);
+
+  it('works', async () => {
+    const pal = new AppTesterPal(getOnboardingRouteForIntent(OnboardingInfoSignupIntent.LOC), {
+      url: '/onboarding/step/4'
+    });
+    const input = await pal.rt.waitForElement(() =>
+      pal.getElement('input', '[name="signupIntent"]')
```

Comment:
This test takes over 1 second to run, I think because it's using a React-loadable component under-the-hood.  It also feels more like an integration test than a unit test, and the logic it's testing feels like it should really be in `onboarding.tsx` but because we want to lazy-load that, I wanted to keep this code outside of that module... hmm.